src/Makefile.am

```diff
@@ -472,15 +472,79 @@ libbitcoinconsensus_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(BOOST_CPPFLAGS) -I$(builddi
 libbitcoinconsensus_la_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
 
 endif
+
+%_wrap.cxx %.py: %.i
+	$(SWIG) -python $(AX_SWIG_PYTHON_CPPFLAGS) $<
+
+
+if ENABLE_BITCOINPY
+
+# libbitcoinpy: compiled with -fPIC to allow pulling into a shared library
+noinst_LIBRARIES = libbitcoinpy.a
+
+#libbitcoinpy_a_LDFLAGS = -g -fPIC -shared $(PYTHON_EXTRA_LDFLAGS) $(AM_LDFLAGS) $(RELDFLAGS)
+libbitcoinpy_a_SOURCES = $(libbitcoin_common_a_SOURCES)
+libbitcoinpy_a_SOURCES += $(libbitcoin_consensus_a_SOURCES)
+libbitcoinpy_a_SOURCES += $(libbitcoin_wallet_a_SOURCES)
+libbitcoinpy_a_SOURCES += $(libbitcoin_server_a_SOURCES)
+libbitcoinpy_a_SOURCES += $(libbitcoin_util_a_SOURCES)
+libbitcoinpy_a_SOURCES += $(crypto_libbitcoin_crypto_a_SOURCES)
+libbitcoinpy_a_CPPFLAGS = $(AM_CPPFLAGS) $(PYTHON_CPPFLAGS) $(libbitcoin_server_a_CPPFLAGS)
+libbitcoinpy_a_CXXFLAGS = $(AM_CPPFLAGS) -fPIC $(PYTHON_CPPFLAGS) $(BITCOIN_INCLUDES)
+
+bitcoinpy_PYTHON_SHARED = _coreio.so _script.so _primitives.so
+bitcoinpy_PYTHON_GENERATED = coreio.py script.py primitives.py
+bitcoinpy_PYTHON_PYC = $(bitcoinpy_PYTHON_GENERATED:.py=.pyc)
+
+noinst_PROGRAMS += $(bitcoinpy_PYTHON_SHARED)
+
+
+PYTHON_MODULE_LINK_OPTIONS=\
+	$(libbitcoinpy_a_LDFLAGS) \
+	-fPIC -shared $(AM_LDFLAGS) $(PYTHON_EXTRA_LDFLAGS) $(RELDFLAGS) $(LIBTOOL_APP_LDFLAGS) \
+	libbitcoinpy.a \
+	./univalue/.libs/libunivalue.a \
+	./secp256k1/.libs/libsecp256k1.a \
+	./leveldb/libleveldb.a \
+	./leveldb/libleveldb_sse42.a \
+	leveldb/libmemenv.a \
+	$(PYTHON_LIBS) $(BOOST_LIBS) $(BDB_LIBS) $(SSL_LIBS) $(CRYPTO_LIBS) $(MINIUPNPC_LIBS) $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS) $(ZMQ_LIBS) -lpthread -lanl
+
+PYTHON_MODULE_CPPFLAGS = $(AM_CPPFLAGS) $(PYTHON_CPPFLAGS) $(libbitcoin_server_a_CPPFLAGS)
+PYTHON_MODULE_CXXFLAGS = -fPIC $(PYTHON_CPPFLAGS) $(AM_CPPFLAGS)
+PYTHON_MODULE_DEPS     = libbitcoinpy.a $(LIBUNIVALUE) $(LIBSECP256K1) $(LIBLEVELDB) $(LIBLEVELDB_SSE42) $(LIBMEMENV)
+
+_coreio_so_SOURCES = coreio_wrap.cxx
+_coreio_so_CPPFLAGS = $(PYTHON_MODULE_CPPFLAGS)
+_coreio_so_CXXFLAGS = $(PYTHON_MODULE_CXXFLAGS)
+_coreio.so$(EXEEXT): $(_coreio_so_OBJECTS) $(PYTHON_MODULE_DEPS)
+	$(CXX) -o $@ $(_coreio_so_OBJECTS) $(PYTHON_MODULE_LINK_OPTIONS)
+
+_script_so_SOURCES = script_wrap.cxx
+_script_so_CPPFLAGS = $(PYTHON_MODULE_CPPFLAGS)
+_script_so_CXXFLAGS = $(PYTHON_MODULE_CXXFLAGS)
+_script.so$(EXEEXT): $(_script_so_OBJECTS) $(PYTHON_MODULE_DEPS)
+	$(CXX) -o $@ $(_script_so_OBJECTS) $(PYTHON_MODULE_LINK_OPTIONS)
+
+_primitives_so_SOURCES = primitives_wrap.cxx
+_primitives_so_CPPFLAGS = $(PYTHON_MODULE_CPPFLAGS)
+_primitives_so_CXXFLAGS = $(PYTHON_MODULE_CXXFLAGS)
+_primitives.so$(EXEEXT): $(_primitives_so_OBJECTS) $(PYTHON_MODULE_DEPS)
+	$(CXX) -o $@ $(_primitives_so_OBJECTS) $(PYTHON_MODULE_LINK_OPTIONS)
+
+endif
+
 #
-CLEANFILES = ($EXTRA_LIBRARIES)
+CLEANFILES = $(EXTRA_LIBRARIES)
 
```

Comment:
good catch. this is #956, right?

Reply:
Yes. I should probably have it in my PR queue here as a separate commit.